src/main/java/games/backgammon/BGGameState.java

```diff
@@ -71,8 +71,16 @@ protected GameType _getGameType() {
      */
     @Override
     protected List<Component> _getAllComponents() {
-        // TODO: add all components to the list
-        return new ArrayList<>();
+        // Let's create one Token per piece (while we're not using Tokens to record the pieces in game state, this may be helpful to something tracking the number of pieces)
+        List<Component> tokens = new ArrayList<>();
+        for (int player = 0; player < 2; player++) {
```

Comment:
player < nplayers :) - also, worth double-checking where this is called, this is not really expected to create components normally, so could potentially create a lot of them with sky-rocketing ID numbers. It might not be used beyond initial setup and/or metrics, but worth a check